sdk/highlight-go/gorm.go

```diff
@@ -0,0 +1,169 @@
+		{cb.Update().After("gorm:update"), p.after(), "after:update"},
+
+		{cb.Row().Before("gorm:row"), p.before("gorm.Row"), "before:row"},
+		{cb.Row().After("gorm:row"), p.after(), "after:row"},
+
+		{cb.Raw().Before("gorm:raw"), p.before("gorm.Raw"), "before:raw"},
+		{cb.Raw().After("gorm:raw"), p.after(), "after:raw"},
+	}
+
+	var firstErr error
+
+	for _, h := range hooks {
+		if err := h.callback.Register("otel:"+h.name, h.hook); err != nil && firstErr == nil {
+			firstErr = fmt.Errorf("callback register %s failed: %w", h.name, err)
+		}
+	}
+
+	return firstErr
+}
+
+func (p *otelPlugin) before(spanName string) gormHookFunc {
+	return func(tx *gorm.DB) {
+		_, tx.Statement.Context = StartTrace(tx.Statement.Context, spanName)
+	}
+}
+
+func (p *otelPlugin) after() gormHookFunc {
+	return func(tx *gorm.DB) {
+		span := trace.SpanFromContext(tx.Statement.Context)
+		defer EndTrace(span)
+
+		attrs := make([]attribute.KeyValue, 0, len(p.attrs)+4)
+		attrs = append(attrs, p.attrs...)
+
+		if sys := dbSystem(tx); sys.Valid() {
+			attrs = append(attrs, sys)
+		}
+
+		vars := tx.Statement.Vars
+
+		var query string
+		if p.excludeQueryVars {
+			query = tx.Statement.SQL.String()
+		} else {
+			query = tx.Dialector.Explain(tx.Statement.SQL.String(), vars...)
+		}
+
+		attrs = append(attrs, semconv.DBStatementKey.String(query))
+		if tx.Statement.Table != "" {
+			attrs = append(attrs, semconv.DBSQLTableKey.String(tx.Statement.Table))
+		}
+		if tx.Statement.RowsAffected != -1 {
+			attrs = append(attrs, dbRowsAffected.Int64(tx.Statement.RowsAffected))
+		}
+
+		span.SetAttributes(attrs...)
+		switch tx.Error {
+		case nil,
+			gorm.ErrRecordNotFound,
+			driver.ErrSkip,
+			io.EOF, // end of rows iterator
+			sql.ErrNoRows:
+			// ignore
+		default:
+			span.RecordError(tx.Error)
+			span.SetStatus(codes.Error, tx.Error.Error())
+		}
+	}
+}
+
+func dbSystem(tx *gorm.DB) attribute.KeyValue {
+	switch tx.Dialector.Name() {
+	case "mysql":
+		return semconv.DBSystemMySQL
+	case "mssql":
+		return semconv.DBSystemMSSQL
+	case "postgres", "postgresql":
+		return semconv.DBSystemPostgreSQL
+	case "sqlite":
+		return semconv.DBSystemSqlite
+	case "sqlserver":
+		return semconv.DBSystemKey.String("sqlserver")
+	case "clickhouse":
+		return semconv.DBSystemKey.String("clickhouse")
+	default:
+		return attribute.KeyValue{}
+	}
+}
+
+// WithGormAttributes configures attributes that are used to create a span.
+func WithGormAttributes(attrs ...attribute.KeyValue) PluginOption {
+	return func(p *otelPlugin) {
+		p.attrs = append(p.attrs, attrs...)
+	}
+}
+
+// WithGormDBName configures a db.name attribute.
+func WithGormDBName(name string) PluginOption {
+	return func(p *otelPlugin) {
+		p.attrs = append(p.attrs, semconv.DBNameKey.String(name))
+	}
+}
+
+// WithoutGormQueryVariables configures the db.statement attribute to exclude query variables
+func WithoutGormQueryVariables() PluginOption {
```

Comment:
are these options something that should be part of the `WithGORMTracing` options?